File: Getting-Practice-with-Python-and-Javascript/1-First-Introductory-Steps/Python/Excercises/program-5.py
# 5-Escribe un programa que solicite al usuario un número entero y luego imprima todos los factores primos de ese número.


def es_primo(num):
    if num < 2:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True

def factores_primos(n):
    factores = []
    for i in range(2, n + 1):
        while es_primo(i) and n % i == 0:
            factores.append(i)
            n /= i
    return factores


userInput = int(input('Ingresa un numero entero: '))
print(f"Los facotres primos de {userInput} son: {factores_primos(userInput)}")

# print(2 ** 0.5)

File: Getting-Practice-with-Python-and-Javascript/1-First-Introductory-Steps/Javascript/Excercises/program-5.js
//   5-Escribe un programa que solicite al usuario un número entero y luego imprima todos los factores primos de ese número.


function esPrimo(num) {
    if (num < 2) {
        return false
    }
    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) return false;
    }
    return true;
}

function factoresPrimos(n) {
    let factores = [];
    for (let i = 2; i <= n; i++) {
        while (esPrimo(i) && n % i === 0) {
            factores.push(i);
            n /= i;
        }
    }
    return factores;
}

let userInput = prompt('Ingresa un numero entero: ');
let numero = parseInt(userInput);

console.log(`Los facotres primos de ${numero} son: ${factoresPrimos(numero)}`)